### PR TITLE
Project Maintenance Tools

### PR DESCRIPTION
**Pull request by nbuchwitz**
Add some more tooling which makes maintenance easier:
- Update README.md based on file contents
- Create GIT tag with changelog as annotation

### CODE REVIEW

**Review by iluminat23 — COMMENTED** [review]
LGTM, but I have not reviewed the tag script.

**Review by RamiGspo — COMMENTED** [review]
In [this commit](https://github.com/RevolutionPi/revpi-hat-data/pull/11/commits/cdcdc1b70149d4c6139faf6cfcf520342e18e637) there is a typo error.
corrent -> current
```
A simple helper script which creates an annotated tag with
the current version schema.
```

File: scripts/update-readme.py [hunk 1]
```diff
@@ -18,11 +18,14 @@
...
         product_id = template_data["pid"] + 100000
+        vendor = template_data["vstr"]
         product_revision = template_data["prev"]
         data_version = template_data["eeprom_data_version"]
         overlay = template_data["dtstr"]
 
-        readme += f"| [{name }]({template}) | PR{product_id} | R{product_revision:02} | {data_version} | {overlay} |\n"
+        readme += f"| [{name }]({template}) | {vendor} | PR{product_id} | R{product_revision:02} | {data_version} | {overlay} |\n"
+
+readme += "\n\n**Pleae note that this README file is auto-generated by the `scripts/update-readme.py` script.**"
```
Comment (RamiGspo) [inline scripts/update-readme.py]: typo error: Pleae -> Please
```
readme += "\n\n**Please note that this README file is auto-generated by the `scripts/update-readme.py` script.**"
```

File: README.md [hunk 2]
```diff
@@ -22,4 +22,4 @@ Device specific JSON templates for RevPi HAT EEPROM generation
 | [RevPi Connect 4 WLAN 32/8GB](./revpi-hat-PR100380R00.json) | KUNBUS GmbH | PR100380 | R00 | 0 | revpi-connect4 |
 
 
-**Pleae note that this README file is auto-generated by the `scripts/update-readme.py` script.**
\ No newline at end of file
+**Pleae note that this README file is auto-generated by the `scripts/update-readme.py` script. More details can be found [here](scripts/README.md)**
```
Comment (RamiGspo) [inline README.md:27]: Pleae -> Please
`**Please note that this README file is auto-generated by the `scripts/update-readme.py` script. More details can be found [here](scripts/README.md)**`

File: scripts/README.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
+# Tools
+
+## Update README (`update-readme.py`)
+
+This script automatically updates the `README.md` file in the repository root.
+The content is build by itterating over files matching `revpi-hat-PR*R*.json` and gathering their product information.
```
Comment (RamiGspo) [inline scripts/README.md:6]: itterating -> iterating

`The content is build by iterating over files matching `revpi-hat-PR*R*.json` and gathering their product information.`

**Review by RamiGspo — APPROVED** [review]
It does what it has to do!
LGTM

Bildschirmfoto vom 2023-04-18 08-46-43

**Pull request merged**